### PR TITLE
Create week14-1.md

### PR DESCRIPTION
**Pull request by jzha123**

### CODE REVIEW

File: docs/en/week14/week14-1.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+<!--- done by Muge --->
+
+## Efficient Inference for Energy-Based Factor Graphs
+
+A Tutorial on Energy-Based Learning (Yann LeCun, Sumit Chopra, Raia Hadsell, Marc’Aurelio Ranzato, and Fu Jie Huang 2006):
+
+Learning and inference with Energy-Based Models involves a minimization of the energy over the set of answers $\mathcal{Y}$ and latent variables $\mathcal{Z}$. When the cardinality of $\mathcal{Y}\times \mathcal{Z}$ is large, this minimization can become intractable. One approach to the problem is to exploit the structure of the energy function in order to perform the minimization efficiently. One case where the structure can be exploited occurs when the energy can be expressed as a sum of individual functions (called factors) that each depend on different subsets of the variables in Y and Z. These dependencies are best expressed in the form of a factor graph. Factor graphs are a general form of graphical models, or belief networks.
+
+<img src="https://i.imgur.com/I56iwGa.png" style="zoom:100%;"/>
```
Comment (Atcold) [inline docs/en/week14/week14-1.md:47]: Images have to be added to the repo. Check out the instructions!
https://atcold.github.io/pytorch-Deep-Learning/en/about/

File: docs/en/week14/week14-1.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+where $Y = [Y_1, Y_2]$ are the output variables and $Z = [Z_1, Z_2]$ are the latent variables. Each factor can be seen as representing soft constraints between the values of its input variables. The inference problem consists in finding:
+
+$$(\bar{Y}, \bar{Z})=\operatorname{argmin}_{y \in \mathcal{Y}, z \in \mathcal{Z}}\left(E_{a}\left(X, z_{1}\right)+E_{b}\left(X, z_{1}, z_{2}\right)+E_{c}\left(z_{2}, y_{1}\right)+E_{d}\left(y_{1}, y_{2}\right)\right)$$
+
+Let’s assume that $Z_1$, $Z_2$, and $Y_1$ are discrete binary variables, and $Y_2$ is a ternary variable. The cardinality of the domain of $X$ is immaterial since X is always observed. The number of possible configurations of $Z$ and $Y$ given X is $2 \times 2 \times 2 \times 3 = 24$. A naive minimization algorithm through exhaustive search would evaluate the entire energy function 24 times (96 single factor evaluations). 
+
+However, we notice that for a given $X$, $E_a$ only has two possible input configurations: $Z_1 = 0$ and $Z_1 = 1$. Similarly, $E_b$ and $E_c$ only have 4 possible input configurations, and $E_d$ has 6. Hence,
+there is no need for more than 2 + 4 + 4 + 6 = 16 single factor evaluations.
```
Comment (Atcold) [inline docs/en/week14/week14-1.md:61]: `$` missing for the math.

File: docs/en/week14/week14-1.md (hunk 1 above)
Comment (jzha123) [reply docs/en/week14/week14-1.md:47]: Hi Alf, 
Sorry I didn't get your point. I can see the image on HackMD, what do you mean add images to the repo? Besides, should 14.md be written by 3 editors for different parts? And how to update _config.yml? Thank you!

File: docs/en/week14/week14-1.md (hunk 2 above)
Comment (jzha123) [reply docs/en/week14/week14-1.md:61]: Hi Alf, 
Could you please refer to where the $ is missing for the math? I didn't find it. Thank you!

File: docs/en/week14/week14-1.md (hunk 1 above)
Comment (jzha123) [reply docs/en/week14/week14-1.md:47]: > Images have to be added to the repo. Check out the instructions!
> https://atcold.github.io/pytorch-Deep-Learning/en/about/

Hi Alf,
Sorry I didn't get your point. I can see the image on HackMD, what do you mean add images to the repo? Besides, should 14.md be written by 3 editors for different parts? And how to update _config.yml? Thank you!

File: docs/en/week14/week14-1.md (hunk 2 above)
Comment (jzha123) [reply docs/en/week14/week14-1.md:61]: > `$` missing for the math.

Hi Alf,
Could you please refer to where the $ is missing for the math? I didn't find it. Thank you!

Comment (Atcold) [reply docs/en/week14/week14-1.md:61]:
```suggestion
there is no need for more than $2 + 4 + 4 + 6 = 16$ single factor evaluations.
```

File: docs/en/week14/week14-1.md (hunk 1 above)
Comment (Atcold) [reply docs/en/week14/week14-1.md:47]: Images have to be uploaded to `docs/images/week14/14-1`. Check the syntax at the link I've sent you. You can also check out previous PRs.

**Comment by Atcold** [comment]
You need to add the summary to your section to the file `14.md` and push that as well.
Finally, the file `docs/_config.yml` needs to reflect the addition of the new files. See what previous weeks look like.

File: docs/en/week14/week14-1.md (hunk 1 above)
Comment (jzha123) [reply docs/en/week14/week14-1.md]: > Images have to be uploaded to `docs/images/week14/14-1`. Check the syntax at the link I've sent you. You can also check out previous PRs.

Hi Alf,

I have uploaded all the images.

**Comment by jzha123** [comment]
> You need to add the summary to your section to the file `14.md` and push that as well.
> Finally, the file `docs/_config.yml` needs to reflect the addition of the new files. See what previous weeks look like.

Yes, I have create a 14.md file and add summay of my part in it.

File: docs/en/week14/14-1.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+For eg: In case of speech/handwriting recognition or natural language translation, the output needs to be grammatically correct and it is not possible to limit the number of output possibilities. The task of the model is to capture the sequential, spatial, or combinatorial structure in the problem domain.
+
+### Early works on structured prediction
+This vector is fed to a TDNN which gives a feature vector which in case of model systems can bee compared to softmax that represents a category. One problem that arises in the case of recognizing the word that was pronounced is different people can pronounce the same word in different ways and speed. To solve this Dynamic Time Warping is used. 
+
+The idea is to provide the system with a set of pre-recorded templates that correspond to sequence or feature vectors that were recorded by someone. The neural network is trained at the same time as the template so that the system learns to recognize the word for different pronunciations. The latent variable allows us to timewarp the feature vector so as to match the length of the templates.
+
+<img src="https://i.imgur.com/T7ajVO7.png" style="zoom:100%;"/>
```
Comment (Atcold) [inline docs/en/week14/14-1.md:19]: This is not pointing to the image on the repo. Please, fix.
Read the instructions!!! https://atcold.github.io/pytorch-Deep-Learning/en/about/

File: docs/en/week14/14-1.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+This can be visualized as a matrix by arrranging the feature vectors from TDNN horizontally and the word templates vertically. Each entry in the matrix corresponds to the distance between the feature vector. This can be visualized as a graph problem where the aim is to start from the bottom left-hand corner and reach the top right corner by traversing the path that minimizes the distance.
+
+To train this latent variable model we need to make the energy for the correct answers as small as possible and larger for every correct answer. To do this we use an objective function that takes in templates for wrong words and pushes them away from the current sequence of features and backpropagates the gradients.
+
+## Energy based factor graphs
+The idea behind energy-based factor graphs is to build an energy-based model in which the energy is sum of partial energy terms or when the probability is a product of factors. The benefit of these models is that efficient inference algorithms can be employed.
+
+<img src="https://i.imgur.com/5YE5pog.png" style="zoom:100%;"/>
```
Comment (Atcold) [inline docs/en/week14/14-1.md:28]: This is not pointing on the image in the repo!

File: docs/en/week14/14-1.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+### Sequence Labeling
+The model takes an input speech signal X and output the labels Y such that the output labels minimize the total energy term.
+<img src="https://i.imgur.com/9k4Frq1.png" style="zoom:100%;"/>
```
Comment (Atcold) [inline docs/en/week14/14-1.md:32]: No image is pointing to the files you just uploaded!!!

File: docs/en/week14/14.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+---
+lang-ref: ch.14
+title: Week 14
+---
+## Lecture part A
+### In this section, we discussed the structured prediction. We first introduced the Energy-Based factor graph and efficient inference for it. Then we gave some examples for simple Energy-Based factor graphs with “shallow” factors. Finally, we discussed the Graph Transformer Net.
```
Comment (Atcold) [inline docs/en/week14/14.md:6]:
```suggestion
In this section, we discussed the structured prediction. We first introduced the Energy-Based factor graph and efficient inference for it. Then we gave some examples for simple Energy-Based factor graphs with “shallow” factors. Finally, we discussed the Graph Transformer Net. 
```

File: docs/en/week14/14-1.md [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+
+### Early works on structured prediction
+This vector is fed to a TDNN which gives a feature vector which in case of model systems can bee compared to softmax that represents a category. One problem that arises in the case of recognizing the word that was pronounced is different people can pronounce the same word in different ways and speed. To solve this Dynamic Time Warping is used. 
+
+The idea is to provide the system with a set of pre-recorded templates that correspond to sequence or feature vectors that were recorded by someone. The neural network is trained at the same time as the template so that the system learns to recognize the word for different pronunciations. The latent variable allows us to timewarp the feature vector so as to match the length of the templates.
+
+<center>
+<img src="{{site.baseurl}}../../images/week14/14-1/Fig1.png" style="zoom: 40%; background-color:#DCDCDC;" /><br>
```
Comment (Atcold) [inline docs/en/week14/14-1.md:20]: This is still wrong. Can you please read the instructions???
https://atcold.github.io/pytorch-Deep-Learning/en/about/

**Comment by Atcold** [comment]
Figures have no captions. Please, add captions (an English description of the image).
Thanks.

**Comment by Atcold** [comment]
Also, remove figure 3 and write the math instead.

**Pull request merged**